krm-functions/lib/interface/v1alpha1/interface.go

```diff
@@ -0,0 +1,240 @@
+func (r *itfce) SetAttachmentType(s string) error {
+
+	// validation -> should be part of the interface api repo
+	switch s {
+	case string(nephioreqv1alpha1.AttachmentTypeNone):
+	case string(nephioreqv1alpha1.AttachmentTypeVLAN):
+	default:
+		return fmt.Errorf("unknown attachmentType")
+	}
+
+	return r.setNestedField(s, attachmentType...)
+}
+
+// SetCNIType sets the cniType in the spec
+func (r *itfce) SetCNIType(s string) error {
+
+	// validation -> should be part of the interface api repo
+	switch s {
+	case string(nephioreqv1alpha1.CNITypeIPVLAN):
+	case string(nephioreqv1alpha1.CNITypeSRIOV):
+	case string(nephioreqv1alpha1.CNITypeMACVLAN):
+	default:
+		return fmt.Errorf("unknown cniType")
+	}
+
+	return r.setNestedField(s, cniType...)
+}
+
+// SetNetworkInstanceName sets the name of the networkInstance in the spec
+func (r *itfce) SetNetworkInstanceName(s string) error {
+	return r.setNestedField(s, networkInstanceName...)
+}
+
+// SetSpec sets the spec attributes in the kubeobject according the go struct
+func (r *itfce) SetSpec(spec *nephioreqv1alpha1.InterfaceSpec) error {
+	if spec == nil {
+		return nil
+	}
+	if spec.AttachmentType != "" {
+		if err := r.SetAttachmentType(string(spec.AttachmentType)); err != nil {
+			return err
+		}
+	} else {
+		if err := r.DeleteAttachmentType(); err != nil {
+			return err
+		}
+	}
+	if spec.CNIType != "" {
+		if err := r.SetCNIType(string(spec.CNIType)); err != nil {
+			return err
+		}
+	} else {
+		if err := r.DeleteCNIType(); err != nil {
+			return err
+		}
+	}
+	if spec.NetworkInstance != nil {
+		if err := r.SetNetworkInstanceName(string(spec.NetworkInstance.Name)); err != nil {
+			return err
+		}
+	} else {
+		return fmt.Errorf("networkInstance is required")
+	}
+	return nil
+}
+
+// DeleteAttachmentType deletes the attachmentType from the spec
+func (r *itfce) DeleteAttachmentType() error {
+	return r.deleteNestedField(attachmentType...)
+}
+
+// DeleteAttachmentType deletes the attachmentType from the spec
+func (r *itfce) DeleteCNIType() error {
+	return r.deleteNestedField(cniType...)
+}
+
+// getStringValue is a generic utility function that returns a string from
+// a string slice representing the path in the yaml doc
+func (r *itfce) getStringValue(fields ...string) string {
+	if r.o == nil {
+		return ""
+	}
+	s, ok, err := r.o.NestedString(fields...)
+	if err != nil {
+		return ""
+	}
+	if !ok {
+		return ""
+	}
+	return s
+}
+
+// setNestedField is a generic utility function that sets a string on
+// a string slice representing the path in the yaml doc
+func (r *itfce) setNestedField(s string, fields ...string) error {
+	if r.o == nil {
+		return fmt.Errorf(errKubeObjectNotInitialized)
+	}
+	if err := r.o.SetNestedField(s, fields...); err != nil {
+		return err
+	}
+	return nil
+}
+
+// deleteNestedField is a generic utility function that deletes
+// a string slice representing the path from the yaml doc
+func (r *itfce) deleteNestedField(fields ...string) error {
```

Comment:
I am of opinion that the `deleteNestedField` method can be removed and the calling function can directly invoke the `r.o.RemoveNestedField` instead by adding a check for `r.o != nil` at the top of the calling method.

Reply:
This is a helper(private) method used in other public methods as part of the refactor.